projects/step-frontend/src/lib/modules/plan/services/artefact.service.ts

```diff
@@ -29,21 +31,23 @@ export class ArtefactService {
       .pipe(
         map((artefactNames: Array<string>) => {
           return artefactNames
-            .filter((artefactName) => this.isSelectable(artefactName))
-            .map((artefactName) => this.getType(artefactName));
+            .filter((artefactName) => this.isSelectable(artefactName) && !!this.getType(artefactName))
+            .map((artefactName) => this.getType(artefactName)!);
         })
       )
       .subscribe((artifacts) => {
         this.availableArtefacts = artifacts;
       });
   }
 
-  private getType(typeName: string): ArtefactType {
+  private getType(typeName: string): ArtefactType | undefined {
     if (this.registry[typeName]) {
       return this.registry[typeName];
-    } else {
-      throw 'Unknown artefact type ' + typeName;
+    } else if (!this.missingArtefactErrorThrown[typeName]) {
+      this.missingArtefactErrorThrown[typeName] = true;
+      throw 'Unknown artefact type "' + typeName + '". Please add it to the type-registry.';
     }
+    return undefined;
```

Comment:
We should not add extensive logic to handle dev errors. If there should be an unknown artefact type it is fine to flood the console with it.

Reply:
Are you sure? I think flooding the console costs a lot of performance and in this case the app is not broken, but there is only an unusable artifact. So if one of our clients ever get into this situations, the app is still usable without any performance issues.

Reply:
Still we would inject extremely specific code to prevent errors that should never happen. If someone creates a custom plugin that triggers this error it is actually a good thing, that he will notice it for sure when looking into the console.